### DISCUSSION TITLE
Why undeteted-chromedriver works?

### DISCUSSION THREAD

**Discussion by huanganding** [ultrafunkamsterdam/undetected-chromedriver#1494] [Q&A]
Why does Undetected-chromedriver work so effectively in bypassing anti-bot detection? Can anyone recommend any books or papers that explain its functionality?

**Comment by fl0aten** [comment]
If I understood it correctly, it is about the cdc variable. Only Chromedriver has this variable, but not Chrome.
Pages can query this variable via JavaScript and know directly if it is a "bot".

undetected-chromedriver patches this variable out and replaces it with its own code. Thus, the browser is no longer directly detectable.

Here is the place where it is patched: https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/master/undetected_chromedriver/patcher.py#L294

Please correct me if i am wrong!
